### PR TITLE
[LogServer] Initial implementation for store path

### PR DESCRIPTION
**Pull request by AhmedSoliman**
[LogServer] Initial implementation for store path

Note 1: This PR contains many todos an various incomplete bits that will be worked out in later PRs. This is highly experimental code and will be refined in the next iterations but I wanted to avoid accumulating code locally.

Note 2: This enables log-server by default, so it's imperative that this isn't included in v1.1 release.

When looking at this, consider the overall design strategy and don't get hang up on the nits or smaller details as they're likely to change anyway.

What this achieves:
- Starts log-server by default (role is now enabled by default)
- Introduces Store/Release network messages for the log-server
- Implements pipelined Store messages from sequencers, no backfill ability.
- LogState management
- Provisional trim infrastructure

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1938).
* #1943
* #1942
* #1939
* __->__ #1938

### CODE REVIEW

**Review by muhamadazmy — APPROVED** [review]
Thanks @AhmedSoliman for the PR. This looks good to me but I left couple of questions

File: crates/log-server/src/rocksdb_logstore/store.rs [hunk 1]
```diff
@@ -85,4 +90,259 @@ impl LogStore for RocksDbLogStore {
             .map_err(RocksDbLogStoreError::from)?;
         Ok(())
     }
+
+    async fn load_loglet_state(
```
[the remark this answers is not in the document]
Comment (AhmedSoliman) [reply crates/log-server/src/rocksdb_logstore/store.rs:94]: It's used in `LogletStateMap::get_or_load(..)` in log-server/src/metadata.rs:76

File: crates/log-server/src/rocksdb_logstore/store.rs [hunk 2]
```diff
@@ -85,4 +90,259 @@ impl LogStore for RocksDbLogStore {
...
+        readopts.set_prefix_same_as_start(true);
+        readopts.set_iterate_lower_bound(oldest_key.to_bytes());
+        // upper bound is exclusive
+        readopts.set_iterate_upper_bound(upper_bound);
+        let mut iterator = self
+            .rocksdb
+            .inner()
+            .as_raw_db()
+            .raw_iterator_cf_opt(&data_cf, readopts);
+        // see to the max key that exists
```
Comment (pcholakov) [inline crates/log-server/src/rocksdb_logstore/store.rs:150]: *seek

**Review by pcholakov — APPROVED** [review]
Very cool! There's a lot going on here, I could definitely spend longer studying this but timeboxing it for now. All makes sense 👍

**Pull request merged**